internal/utils/arrays.go

```diff
@@ -0,0 +1,15 @@
+package utils
+
+func Map[T any, O any](input []T, mapper func(T) O) []O {
```

Comment:
Suggestion: A comment explaining the usage would go a long way. Seems like it's a general Map function that accepts any type of input and calls the mapper function passed it?